source/common/network/io_socket_handle_impl.cc

```diff
@@ -517,13 +530,17 @@ Address::InstanceConstSharedPtr IoSocketHandleImpl::peerAddress() {
   return Address::addressFromSockAddr(ss, ss_len);
 }
 
-Event::FileEventPtr IoSocketHandleImpl::createFileEvent(Event::Dispatcher& dispatcher,
-                                                        Event::FileReadyCb cb,
-                                                        Event::FileTriggerType trigger,
-                                                        uint32_t events) {
-  return dispatcher.createFileEvent(fd_, cb, trigger, events);
+void IoSocketHandleImpl::initializeFileEvent(Event::Dispatcher& dispatcher, Event::FileReadyCb cb,
+                                             Event::FileTriggerType trigger, uint32_t events) {
+  ASSERT(file_event_ == nullptr, "Attempting to initialize two `file_event_` for the same "
+                                 "file descriptor. This is not allowed.");
+  file_event_ = dispatcher.createFileEvent(fd_, cb, trigger, events);
 }
 
+void IoSocketHandleImpl::activateFileEvents(uint32_t events) { file_event_->activate(events); }
+
+void IoSocketHandleImpl::enableFileEvents(uint32_t events) { file_event_->setEnabled(events); }
```

Comment:
I'm worried about the crash potential of these calls without nullptr checks around them.  Could you change these two methods to:

if (file_event_) {
  file_event_->METHOD_NAME(events);
} else {
  ENVOY_BUG(false, "Null file_event_");
}

Reply:
added the null check in 13ff61610c48391cf70ae52dbc51b2d88c2132dd